yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -553,21 +708,43 @@ func (p *ParserIssueDetector) ParseAndProcessDDL(query string) error {
 			}
 		}
 
-		// Collect the foreign key constraint details from CREATE TABLE statement.
-		// These are stored temporarily and will be processed later in FinalizeColumnMetadata,
-		// once all tables and columns are parsed and available in columnMetadata.
+		// Track constraints for PK/UK and column-level NOT NULL
 		for _, constraint := range table.Constraints {
-			if constraint.ConstraintType != queryparser.FOREIGN_CONSTR_TYPE {
-				continue
+			switch constraint.ConstraintType {
```

Comment:
nit: you could add helper method(s) to TableMetadata to `addConstraint(type, name, columns)`